### PR TITLE
BAEL-1112 - Apache Tapestry

### PR DESCRIPTION
**Pull request by SmartyAnsh**

### CODE REVIEW

File: apache-tapestry/pom.xml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    <build>
+        <finalName>apache-tapestry</finalName>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>2.3.2</version>
+                <configuration>
+                    <source>1.6</source>
+                    <target>1.6</target>
```
Comment (KevinGilmore) [inline apache-tapestry/pom.xml]: Unless there's a reason not to, we should be using 1.8 (Java 8).

File: apache-tapestry/pom.xml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        </dependency>
+
+        <!-- Include the Log4j implementation for the SLF4J logging framework -->
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-log4j12</artifactId>
+            <version>${slf4j-release-version}</version>
+        </dependency>
+
+
```
Comment (KevinGilmore) [inline apache-tapestry/pom.xml]: one blank line is plenty

File: apache-tapestry/pom.xml [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            <url>http://repository.jboss.org/nexus/content/groups/public/</url>
+        </repository>
+
+        <!-- This repository is only needed when the Tapestry version is a preview release, rather
+             than a final release. -->
+        <repository>
+            <id>apache-staging</id>
+            <url>https://repository.apache.org/content/groups/staging/</url>
+        </repository>
+    </repositories>
```
Comment (KevinGilmore) [inline apache-tapestry/pom.xml]: indentation is off in this section

File: apache-tapestry/pom.xml (hunk 1 above)
Comment (SmartyAnsh) [reply apache-tapestry/pom.xml]: My bad, it should be 1.8 - Fixed

File: apache-tapestry/pom.xml (hunk 2 above)
Comment (SmartyAnsh) [reply apache-tapestry/pom.xml]: Fixed

File: apache-tapestry/pom.xml (hunk 3 above)
Comment (SmartyAnsh) [reply apache-tapestry/pom.xml]: Fixed

**Comment by SmartyAnsh** [comment]
Fixed PR reviews

File: apache-tapestry/src/main/java/com/baeldung/tapestry/pages/Login.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    @InjectComponent
+    private Form login;
+
+    @Property
+    private String email;
+
+    @Property
+    private String password;
+
+    void onValidateFromLogin() {
```
Comment (eric-martin) [inline apache-tapestry/src/main/java/com/baeldung/tapestry/pages/Login.java]: I think this should be onValidateFormLogin

File: apache-tapestry/src/main/java/com/baeldung/tapestry/pages/Login.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        if(email == null || password == null) {
+            alertManager.error("Email/Password is null");
+            login.recordError("Validation failed");
+        }
+    }
+
+    Object onSuccessFromLogin() {
```
Comment (eric-martin) [inline apache-tapestry/src/main/java/com/baeldung/tapestry/pages/Login.java]: onSuccessFormLogin

File: apache-tapestry/src/main/java/com/baeldung/tapestry/pages/Login.java (hunk 4 above)
Comment (SmartyAnsh) [reply apache-tapestry/src/main/java/com/baeldung/tapestry/pages/Login.java]: this is correct. The convention of the method name is on**Event**From**Component**

File: apache-tapestry/src/main/resources/com/baeldung/tapestry/components/Layout.tml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<html xmlns:t="http://tapestry.apache.org/schema/tapestry_5_4.xsd">
+	<head>
+	    <title>${title}</title>
```
Comment (KevinGilmore) [inline apache-tapestry/src/main/resources/com/baeldung/tapestry/components/Layout.tml]: seems to be using a combination of spaces and tabs to indent; let's stick to spaces

File: apache-tapestry/pom.xml [hunk 7]
```diff
@@ -0,0 +1,147 @@
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.baeldung</groupId>
+    <artifactId>apache-tapestry</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+    <packaging>war</packaging>
+    <name>apache-tapestry Tapestry 5 Application</name>
```
[the remark this answers is not in the document]
Comment (SmartyAnsh) [reply apache-tapestry/pom.xml]: fixed

File: apache-tapestry/src/main/resources/com/baeldung/tapestry/components/Layout.tml (hunk 6 above)
Comment (SmartyAnsh) [reply apache-tapestry/src/main/resources/com/baeldung/tapestry/components/Layout.tml]: fixed the indentation

**Pull request merged**